### PR TITLE
Get events support subcription

### PR DESCRIPTION
**Pull request by Lin1404**

### CODE REVIEW

File: src/main/java/com/timebook/timebook/service/EventService.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+public class EventService {
+    private final UserService userService;
+    private final EventRepository eventRepository;
+
+    public EventService(EventRepository eventRepository, UserService userService) {
+        this.eventRepository = eventRepository;
+        this.userService = userService;
+    }
+
+    public Event post(Event requestEvent) {
```
Comment (HaofengLiang) [inline src/main/java/com/timebook/timebook/service/EventService.java]: Maybe just call it save.

File: src/main/java/com/timebook/timebook/service/EventService.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                .collect(Collectors.toList());
+        for (int i = 0; i < subscritionList.size(); i++) {
+            String tragetEmail = subscritionList.get(i);
+            List<Event> subEvents = eventRepository.findAllByEmail(tragetEmail).stream().filter(datefilter)
+                    .collect(Collectors.toList());
+            subscription.put(tragetEmail, subEvents);
+        }
+
+        result.put(userEmail, userEvents);
+        result.put("Subscription", subscription);
```
Comment (HaofengLiang) [inline src/main/java/com/timebook/timebook/service/EventService.java]: Using all lower case key may reduce some amount of issues.

File: src/main/java/com/timebook/timebook/service/EventService.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    public Event post(Event requestEvent) {
+        return eventRepository.save(requestEvent);
+    }
+
+    public void delete(long id) {
+        eventRepository.deleteById(id);
+    }
+
+    public Predicate<Event> datefilter(String period, String date) {
```
Comment (HaofengLiang) [inline src/main/java/com/timebook/timebook/service/EventService.java]: Maybe name it to createDateFilter. (This is actually a kind of factory pattern.)

File: src/main/java/com/timebook/timebook/service/EventService.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        List<String> subscritionList = fromUser.getSubscriptions().stream().map(User::getEmail)
+                .collect(Collectors.toList());
+
+        JSONObject result = new JSONObject();
+        JSONObject subscription = new JSONObject();
+
+        Predicate<Event> datefilter = this.datefilter(period, date);
+        List<Event> userEvents = eventRepository.findAllByEmail(userEmail).stream().filter(datefilter)
+                .collect(Collectors.toList());
+        for (int i = 0; i < subscritionList.size(); i++) {
```
Comment (HaofengLiang) [inline src/main/java/com/timebook/timebook/service/EventService.java]: nit: Use foreach loop or foreach method might reduce lines and make it easier to read.

**Review by HaofengLiang — CHANGES_REQUESTED** [review]

File: src/main/java/com/timebook/timebook/service/EventService.java (hunk 1 above)
Comment (Lin1404) [reply src/main/java/com/timebook/timebook/service/EventService.java]: Because this API included 'add' and 'update' events

**Review by HaofengLiang — APPROVED** [review]

**Pull request merged**